examples/plot_glm_retinal_ganglia.py

```diff
@@ -0,0 +1,289 @@
+# ``spike_times`` (recorded time of the spikes)
+
+from pyglmnet.datasets import fetch_RGCs_data
+
+with TemporaryDirectory(prefix="tmp_glm-tools") as temp_dir:
+    dpath = fetch_RGCs_data(dpath=temp_dir, accept_rgcs_license=True)
+    with open(op.join(dpath, 'data_RGCs.json'), 'r') as f:
+        rgcs_dataset = json.loads(f.read())
+
+stim = np.array(rgcs_dataset['stim'])
+stim_times = np.array(rgcs_dataset['stim_times'])
+
+# spike times for all 4 cells (0-1 are OFF cells, 2-3 are ON cells)
+spike_times = [
+    np.array(rgcs_dataset['spike_times']['cell_0']),
+    np.array(rgcs_dataset['spike_times']['cell_1']),
+    np.array(rgcs_dataset['spike_times']['cell_2']),
+    np.array(rgcs_dataset['spike_times']['cell_3']),
+]
+
+n_cells = len(spike_times)  # total number of cells
+dt = stim_times[1] - stim_times[0]  # time between the stimulation
+n_t = len(stim)  # total number of the stimulation
+f = 1. / dt  # frequency of the stimulation
+
+########################################################
+#
+# You can pick a cell to work with and visualize the spikes for one second.
+# In this case, we will pick cell number 2 (ON cell).
+
+cell_idx = 2  # pick cell number 2 (ON cell)
+spike_time = spike_times[cell_idx]  # pick spike time to work with
+n_spikes = len(spike_time)  # number of spikes
+
+print('Loaded RGC data: cell {}'.format(cell_idx))
+print('Number of stim frames: {:d} ({:.1f} minutes)'.
+      format(n_t, n_t * dt / 60))
+print('Time bin size: {:.1f} ms'.format(dt * 1000))
+print('Number of spikes: {} (mean rate = {:.1f} Hz)\n'.
+      format(n_spikes, n_spikes / n_t * 60))
+
+# sample indices for visualization
+sample_index = np.arange(120)
+t_sample = dt * sample_index
+
+plt.subplot(2, 1, 1)
+plt.step(t_sample, stim[sample_index])
+plt.title('Raw Stimulus (full field flicker)'
+          ' and corresponding spike times')
+plt.ylabel('Stimulation Intensity')
+
+plt.subplot(2, 1, 2)
+tspplot = spike_time[(spike_time >= t_sample.min()) &
+                     (spike_time < t_sample.max())]
+plt.stem(spike_time[sample_index],
+         [1] * len(spike_time[sample_index]))
+plt.xlim([t_sample.min(), t_sample.max()])
+plt.ylim([0, 2])
+plt.xlabel('Time (s)')
+plt.ylabel('Spikes (0 or 1)')
+plt.show()
+
+########################################################
+#
+# We can use ``scipy``'s function ``hankel`` to make our design matrix.
+
+n_t_filt = 25  # tweak this to see different results
+stim_padded = np.pad(stim, (n_t_filt - 1, 0))
+Xdsgn = hankel(stim_padded[0: -n_t_filt + 1], stim[-n_t_filt:])
+
+plt.imshow(Xdsgn[:50, :],
+           cmap='binary',
+           aspect='auto',
+           interpolation='nearest')
+plt.xlabel('lags before spike time')
+plt.ylabel('time bin of response')
+plt.title('Sample first 50 rows of design'
+          ' matrix created using Hankel')
+plt.colorbar()
+plt.show()
+
+########################################################
+# **Compute Spike-Triggered Average (STA)**
```

Comment:
This is very neuro stuff and not used later on in the tutorial, can we remove this?